### PR TITLE
formik/ENG-2819: refactor Roles screen to formik

### PR DESCRIPTION
**Pull request by gReis89**

### CODE REVIEW

File: src/state/roles/actions.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ export const fetchRole = roleCode => dispatch =>
       response.json().then((data) => {
         if (response.ok) {
           dispatch(initialize('role', data.payload));
```
Comment (ichalagashvili) [inline src/state/roles/actions.js:68]: do we still need `initialize` ? since redux-form is removed I dont see any reason to call it again?

Comment (gReis89) [reply src/state/roles/actions.js:68]: this same action is being used on other pages as well, so I kept it to ensure I wouldn't break anything out of my scope

File: src/ui/roles/common/RoleForm.js [hunk 2]
```diff
@@ -27,154 +25,198 @@ const msgs = defineMessages({
...
+  const isEdit = mode === EDIT_MODE;
+  const superuserToggled = values.permissions ? values.permissions.superuser : false;
 
-  handleToggleSuperuser(superuserToggled) {
-    const { onToggleSuperuser, permissions } = this.props;
-    onToggleSuperuser({ superuserToggled, permissions });
-  }
+  useEffect(() => {
+    onWillMount(roleCode);
+  }, [roleCode, onWillMount]);
```
Comment (raxkaynan) [inline src/ui/roles/common/RoleForm.js]: This is just trivial but I think the `onWillMount` naming seems to refer more closely to the deprecated lifecycle method. Perhaps a better one would be `onDidMount` or simply just `onMount`?

**Review by raxkaynan — CHANGES_REQUESTED** [review]
Just a minor thing.

**Review by jeffgo10 — APPROVED** [review]

**Pull request merged**